### PR TITLE
MINIFICPP-852: Add details to restful response. CoAP will not include…

### PR DESCRIPTION
**Pull request by phrocker**
… details for the time being

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/http-curl/tests/C2FailedUpdateTest.cpp [hunk 1]
```diff
@@ -57,8 +58,30 @@ class ConfigHandler : public CivetHandler {
...
+    long long contentlen = req_info->content_length;
+    char buf[1024];
+
+    std::string data;
+    while (readlen < contentlen) {
+      remainlen = contentlen - readlen;
+      if (remainlen > sizeof(buf)) {
+        remainlen = sizeof(buf);
+      }
+      remainlen = mg_read(conn, buf, (size_t) contentlen);
```
Comment (arpadboda) [inline extensions/http-curl/tests/C2FailedUpdateTest.cpp]: I think this was meant to be:
```
      remainlen = mg_read(conn, buf, (size_t) remainlen);
      if (remainlen <= 0) {
        break;
      }
      readlen += remainlen;
```

File: libminifi/include/core/logging/LoggingInterceptor.h [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+class Interceptor {
+ public:
+  void intercept_error(const std::string &msg) {
+    std::lock_guard<std::mutex> lock(guard_);
+    errors_.emplace_back(msg);
+  }
+
+  const std::vector<std::string> &getErrors() const {
+    return errors_;
```
Comment (arpadboda) [inline libminifi/include/core/logging/LoggingInterceptor.h]: I'm a bit afraid of this. 
When errors are logged the intercept_error func above will lock the mutex, although whoever calling this function will get a reference to the vector without knowing about the mutex. 
I would prefer lock and copy here.  A match function (to be able to search for specific errors using regex) would be nice, too - that can do the job using the lock and avoid copies.

File: libminifi/include/core/logging/LoggingInterceptor.h [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+class LoggingInterceptor : public Interceptor {
+
+ public:
+
+  LoggingInterceptor() {
+    LoggingInterceptors::getInstance()->addInterceptor(this);
+  }
+
+  ~LoggingInterceptor() {
+    std::lock_guard<std::mutex> lock(guard_);
```
Comment (arpadboda) [inline libminifi/include/core/logging/LoggingInterceptor.h]: This is a possible deadlock. 
LoggingInterceptor is being destructed while LoggingInterceptors::intercept_error is being executed. 
Both of these grab's it's own lock before trying to call a member of the other to lock that, too.

File: libminifi/include/core/logging/LoggingInterceptor.h (hunk 2 above)
Comment (phrocker) [reply libminifi/include/core/logging/LoggingInterceptor.h]: That's a good catch. This was unintentional

File: libminifi/include/core/logging/LoggingInterceptor.h (hunk 3 above)
Comment (phrocker) [reply libminifi/include/core/logging/LoggingInterceptor.h]: Yah that's a bad design, will fix. Though in thinking about it the logging interceptor is not what I want to do. So I will likely remove it entirely.  It should be more purposeful.n

Comment (phrocker) [reply libminifi/include/core/logging/LoggingInterceptor.h]: Errors should be captured mode purposefully, rather n

**Pull request closed without merging**